packages/browser-wallet/src/background/identity-issuance.ts

```diff
@@ -12,85 +12,106 @@ import { confirmIdentity } from './confirmation';
 
 const redirectUri = 'ConcordiumRedirectToken';
 const codeUriKey = 'code_uri=';
+const IDP_ALARM_NAME = 'idp';
 
-function handleExternalIssuance(url: string, respond: (response: IdentityIssuanceBackgroundResponse) => void) {
-    chrome.tabs
-        .create({
-            url,
-        })
-        .then((tab) => {
-            const closedListener = (tabId: number) => {
-                if (tabId === tab.id) {
-                    chrome.tabs.onRemoved.removeListener(closedListener);
-                    respond({
-                        status: BackgroundResponseStatus.Aborted,
-                    });
-                }
-            };
-            chrome.tabs.onRemoved.addListener(closedListener);
-
-            const onComplete = new Promise<string>((resolve) => {
-                chrome.webRequest.onBeforeRedirect.addListener(
-                    function redirectListener(details) {
-                        if (details.redirectUrl.includes(redirectUri)) {
-                            chrome.webRequest.onBeforeRedirect.removeListener(redirectListener);
-                            resolve(details.redirectUrl);
-                        }
-                    },
-                    { urls: ['<all_urls>'], tabId: tab.id }
-                );
-                chrome.webRequest.onBeforeRequest.addListener(
-                    function requestListener(details) {
-                        if (details.url.includes(redirectUri)) {
-                            chrome.webRequest.onBeforeRequest.removeListener(requestListener);
-                            resolve(details.url);
-                        }
-                    },
-                    { urls: ['<all_urls>'], tabId: tab.id }
-                );
-            });
-            onComplete.then((redirectUrl) => {
-                chrome.tabs.onRemoved.removeListener(closedListener);
-                if (tab.id !== undefined) {
-                    chrome.tabs.remove(tab.id);
-                }
-                respond({
-                    status: BackgroundResponseStatus.Success,
-                    result: redirectUrl.substring(redirectUrl.indexOf(codeUriKey) + codeUriKey.length),
-                });
-            });
-        });
-}
+const respond = async (response: IdentityIssuanceBackgroundResponse) => {
+    // TODO this hack is only needed due to a bug in chrome, which prevents chrome.webRequest hooks from waking up the service worker from inactive state.
+    // Bug will be fixed in chrome v107
+    // https://bugs.chromium.org/p/chromium/issues/detail?id=1024211#c73
+    chrome.alarms.clear(IDP_ALARM_NAME);
 
-async function startIdentityIssuance({
-    baseUrl,
-    ...identityRequestInputs
-}: IdentityRequestInput & { baseUrl: string }) {
     const network = await storedCurrentNetwork.get();
```

Comment:
Previously the network was loaded once and given as static input to the `respond` function. The issue with loading it dynamically each time is that the network could have changed, and you end up putting the identity into the wrong network.

Reply:
It might suffice, but I dislike that you have two values now that can be updated separately. I think the better short-term solution would be to have a single session value that contains the pending identity _and_ the network it is for. We really need to more sure that this is super robust and that we don't end up mixing network keys in the future.

What do you think @shjortConcordium?

Reply:
Yeah, I don't see a reason not to mash the session values together.

Reply:
I guess the only thing that makes me doubt the robustness is that you can keep the tab open while switching network, but I guess we should make sure to reset the tabId when switching network. If we don't already.

Reply:
> I guess the only thing that makes me doubt the robustness is that you can keep the tab open while switching network, but I guess we should make sure to reset the tabId when switching network. If we don't already.

As I see it, this won't be an issue when we store the network with the pending identity, as that will also be the network used to store the identity.

Reply:
Consolidated the pending identity props into one entry in sessions storage now.